CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+cmake_minimum_required( VERSION 3.1 )
+
+project( ADiKtED LANGUAGES C )
+set(PROJECT_DESCRIPTION "Dungeon Keeper 1 map editor")
+
+include(GNUInstallDirs)
+include(FindPkgConfig)
+
+if( NOT CMAKE_BUILD_TYPE )
+	set( CMAKE_BUILD_TYPE Release CACHE STRING "Choose the type of build." FORCE )
+endif()
+
+set( CMAKE_C_STANDARD 99 )
+set( CMAKE_C_STANDARD_REQUIRED ON )
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wno-switch -Wno-unused-parameter -Wno-unused-result" )
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pedantic" )
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wextra" )
+set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror-implicit-function-declaration -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare" )
+
+file( GLOB_RECURSE EXAMPLES_PUTGEMS_SOURCES examples/putgems/*.c
+	examples/putgems/*.h )
+file( GLOB_RECURSE EXAMPLES_PUTTRAIN_SOURCES examples/puttrain/*.c
+	examples/puttrain/*.h )
+file( GLOB_RECURSE EXAMPLES_VIEWMAP_SOURCES examples/viewmap/*.c
+	examples/viewmap/*.h )
+file( GLOB_RECURSE EXAMPLES_PUTEMPLE_SOURCES examples/putemple/*.c
+	examples/putemple/*.h )
+
+find_package( SDL REQUIRED SDL::Main )
```

Comment:
Consider making it optional so if SDL is not found the `viewmap` and `putemple` example projects are not built. Not super important though.